recipes/libdwarf/all/conanfile.py

```diff
@@ -1,84 +1,103 @@
-from conans import ConanFile, CMake, tools
-import functools
+from conan import ConanFile
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rmdir, rename
+from conan.tools.build import cross_building
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
 import os
 
-required_conan_version = ">=1.33.0"
-
+required_conan_version = ">=1.53.0"
 
 class LibdwarfConan(ConanFile):
     name = "libdwarf"
     description = "A library and a set of command-line tools for reading and writing DWARF2"
-    topics = ("libdwarf", "dwarf2", "debugging", "dwarf")
+    license = ("LGPL-2.1-only", "GPL-2.0-only", "BSD-2-Clause-Views")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.prevanders.net/dwarf.html"
-    license = "LGPL-2.1"
+    topics = ("debug", "dwarf", "dwarf2", "elf")
 
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "with_dwarfgen": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
+        "with_dwarfgen": False,
     }
 
-    generators = "cmake"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
```

Comment:
You should probably define the license here as only `"LGPL-2.1-only"` when the library is built without `dwarfgen`.

Reply:
@jwillikers 
Thank you for your advices!
I will fix it.

Reply:
@jwillikers 
Fixed.